apps/challenges/models.py

```diff
@@ -80,6 +84,13 @@ def is_active(self):
         return False
 
 
+@receiver(post_save, sender=Challenge)
+def publish_message_on_change(instance, created=False, **kwargs):
```

Comment:
I think that this PR is missing a very important thing. This signal handler will be called when every any field of model instance is updated and corresponding to it, a message will be published each time. If we clearly see that the worker only needs challenge id (which will never change) to maintain map, and then corresponding files, we come to a realization that the worker should only be notified of changes in file field. 

This way we prevent ourselves and the worker from unnecessarily loading files again and again.